src/buildstream_plugins/sources/docker.py

```diff
@@ -329,6 +328,10 @@ class DockerSource(Source):
 
     BST_MIN_VERSION = "2.0"
 
+    # url is deprecated, but accept it as a valid key so that we can raise
+    # a nicer warning.
+    CONFIG_KEYS = ["architecture", "registry-url", "image", "os", "ref", "track", "url"] + Source.COMMON_CONFIG_KEYS
```

Comment:
So I see you added "architecture" here: does this affect the cache keys, or does it only affect tracking?

BTW, moving this here departs from the pattern used by other plugins in this repo.

Reply:
Oh, it's "architecture" and "os". But the same applies.

Reply:
It's actually *old* parameter. This plugin has been incorrectly not allowing it to be used.

Reply:
It does not affect cache key. It affects tracking.

Reply:
These parameters have existed way back in bst_external. This config keys bug existed there too. This plugin has been ported bug for bug into this repo.

Reply:
Fixed.